solidity/contracts/AugurConstantProductMarketRouter.sol

```diff
@@ -436,19 +439,15 @@ contract AugurConstantProductRouter {
 	}
 
 	function getUserLpTokenIdsForMarket(address augurMarketAddress, address user) external view returns (uint256[] memory) {
-		uint256[] memory marketLpTokenIds = lpTokenIds[augurMarketAddress];
-		uint256 length = 0;
-		for (uint256 i = 0; i < marketLpTokenIds.length; i++) {
-			if (IPositionManager(Constants.UNIV4_POSITION_MANAGER).ownerOf(marketLpTokenIds[i]) == user) length++;
-		}
-		uint256[] memory userLpTokenIds = new uint256[](length);
-		for (uint256 i = 0; i < marketLpTokenIds.length; i++) {
-			if (IPositionManager(Constants.UNIV4_POSITION_MANAGER).ownerOf(marketLpTokenIds[i]) == user) {
-				length--;
-				userLpTokenIds[length] = marketLpTokenIds[i];
-			}
+		return lpTokenIds[user][augurMarketAddress];
+	}
+
+	function userOwnsLpToken(address augurMarketAddress, address user, uint256 lpToken) internal view returns (bool) {
+		uint256[] memory userMarketLpTokens = lpTokenIds[user][augurMarketAddress];
+		for (uint256 i = 0; i < userMarketLpTokens.length; i++) {
+			if (userMarketLpTokens[i] == lpToken) return true;
 		}
-		return userLpTokenIds;
+		return false;
 	}
```

Comment:
What is the point of this function?  Doesn't seem too useful now that you can just do `lpTokenIds(user, augurMarketAddress)` and look through the (short) list yourself off-chain?

Reply:
This is an internal function used to prevent non owners from using functions that modify the associated liquidity, not an external view